Add unit tests for ads manager CTA in catalog overview

ianlin opened the pull request:
### Changes proposed in this Pull Request:

Closes #280.

This PR adds a unit test for Pinterest ads manager CTA under catalog overview.


### Screenshots:

N/A.


### Detailed test instructions:

1. Checkout this branch `feature/280-add-unit-tests-ads-manager-cta`.
2. Run `npm run test-unit`.
3. `assets/source/catalog-sync/sections/SyncState.test.js` is covered in the test.

ianlin committed:
Add coverage in .gitignore

ianlin committed:
Add @testing-library/jest-dom module for testing

So we can use more jest matchers like `toHaveTextContent`.

ianlin committed:
Add a required prop `caption` to TablePlaceholder component

Otherwise running tests would have warnings and fail

ianlin committed:
Add tests for ads manager CTA in catalog overview

puntope commented on assets/source/catalog-sync/sections/SyncState.test.js:
```diff
@@ -0,0 +1,26 @@
+
+		expect( getByRole( 'heading' ) ).toHaveTextContent( 'Overview' );
+
+		expect(
+			getByText( 'Set up and manage ads to increase your reach with' )
+		).toBeTruthy();
```
💅 

Since we're testing in L22 already the text inside the link, I would remove the extra query here testing the rest of the text.

ianlin replied:
You're right, fixed in b3e42d2, thanks!

puntope commented on assets/source/catalog-sync/sections/SyncState.test.js:
```diff
@@ -0,0 +1,26 @@
+		).toBeTruthy();
+
+		expect( getByRole( 'link' ) ).toHaveTextContent(
+			'Pinterest ads manager'
```
💅

I would add `Pinterest ads manager(opens in a new tab)` to test as well if the link is ExternalLink.

ianlin replied:
Spot on! Fixed in 424f5ec, thanks.

puntope reviewed (approved):
✅  LGTM, thanks for adding tests!

- I was able to successfully run the test in both IDE and by npm. 👍 
- Removing the link made the test fail 👍 

💅   

- If I change ExternalLink into Link - Test is not failing
- Maybe omit extra query asking again for text in the Footer (see code comments)
- Since we added Head test btw. We could also add Summary and Table as well. 

```
test( 'should render Sync Summary', () => {
		const { getAllByTestId } = render( <SyncState /> );

		expect( getAllByTestId( 'summary-placeholder' ) ).toBeTruthy();
	} );

	test( 'should render Sync Table', () => {
		const { getByText } = render( <SyncState /> );

		expect( getByText( 'Property' ) ).toBeTruthy();
	} );
```

ianlin committed:
Remove an unnecessary test on catalog overview's footer

Since we've already tested the text in the link, tests for the rest of
the text can be safely removed.

ianlin committed:
Update test to ensure link is ExternalLink in catalog overview footer

ianlin committed:
Add tests for SyncStateSummary and SyncStateTable in catalog overview

ianlin commented:
> Since we added Head test btw. We could also add Summary and Table as well.

@puntope Thanks for the code example, added in c980737. Now I know we can use a test ID for testing in RTL, thanks.

puntope commented:
🥳  This was already good for merge but now is perfect! Great work going one step ahead.

> Now I know we can use a test ID for testing in RTL, thanks.

Notice this should be the last option since it requires adding markup in our code. In this case, it was already there, so perfect!
